src/Development/IDE/Core/Compile.hs

```diff
@@ -148,11 +160,55 @@ typecheckModule (IdeDefer defer) hsc pm = do
     where
         demoteIfDefer = if defer then demoteTypeErrorsToWarnings else id
 
+modifyPLS :: DynLinker -> (PersistentLinkerState -> IO (PersistentLinkerState, a)) -> IO a
+modifyPLS dl f =
+  modifyMVar (dl_mpls dl) (fmapFst pure . f . fromMaybe undefined)
+  where fmapFst f = fmap (\(x, y) -> (f x, y))
+
+
+-- | Need to uninterruptibleMask while communicating with `ghc-iserv`
+maskHooks :: DynFlags -> DynFlags
+maskHooks df =
+  df { hooks = (hooks df) { runMetaHook = Just $ modifyMetaHook (runMetaHook $ hooks df)
+                          , hscCompileCoreExprHook = Just $ hscCompileCoreExprMasked } }
+  where
+    modifyMetaHook :: Maybe (MetaHook TcM) -> (MetaHook TcM)
+    modifyMetaHook mh r e =
+      uninterruptibleMaskM_ $
+        maybe (defaultRunMeta r e) (\f -> f r e) mh
+
+
+hscCompileCoreExprMasked :: HscEnv -> SrcSpan -> CoreExpr -> IO ForeignHValue
+hscCompileCoreExprMasked hsc_env srcspan ds_expr
+    = do { let dflags = hsc_dflags hsc_env
+
+           {- Simplify it -}
+         ; simpl_expr <- simplifyExpr dflags ds_expr
+
+           {- Tidy it (temporary, until coreSat does cloning) -}
+         ; let tidy_expr = tidyExpr emptyTidyEnv simpl_expr
+
+           {- Prepare for codegen -}
+         ; prepd_expr <- corePrepExpr dflags hsc_env tidy_expr
+
+           {- Lint if necessary -}
+         ; lintInteractiveExpr "hscCompileExpr" hsc_env prepd_expr
```

Comment:
skip?

Reply:
I guess we can. I copied the entire function verbatim from `GHC` except for the `uninteruptibleMask` around `linkExpr`.

The function is pretty harmless anyway:

```haskell
lintInteractiveExpr what hsc_env expr
  | not (gopt Opt_DoCoreLinting dflags)
  = return ()
```